public/js/index_datatables.js

```diff
@@ -46,6 +46,20 @@ IndexTable.initializeAll = function () {
     $.fn.dataTableExt.oStdClasses.sStripeOdd = "gtr0";
     $.fn.dataTableExt.oStdClasses.sStripeEven = "gtr1";
 
+    // set custom columns
+    let columns = [];
+    let columnCount = localStorage.columnCount ? parseInt(localStorage.columnCount) : 2;
+    for (let i = 1; i <= columnCount; i++) {
+        columns.push({
+            data: "tags",
+            className: `customheader${i} itd`,
+            name: localStorage[`customColumn${i}`] || `defaultCol${i}`,
+            render: (data, type) => IndexTable.renderColumn(localStorage[`customColumn${i}`], type, data)
+        });
+    }
+    columns.unshift({ data: null, className: "title itd", name: "title", render: IndexTable.renderTitle });
```

Comment:
I'd just initialize `columns` with this line already in instead of unshifting it in after adding all the custom headers.